src/idl_gen_cpp.cpp

```diff
@@ -233,6 +233,28 @@ class CppGenerator : public BaseGenerator {
     return keywords_.find(name) == keywords_.end() ? name : name + "_";
   }
 
+  std::string Name(const FieldDef &field) const {
+    // the union type field suffix is immutable.
+    static size_t union_suffix_len = strlen(UnionTypeFieldSuffix());
+    const bool is_union_type = field.value.type.base_type == BASE_TYPE_UTYPE;
+    // early return if no case transformation required
+    if (opts_.cpp_object_api_field_case_style ==
+        IDLOptions::CaseStyle_Unchanged)
+      return EscapeKeyword(field.name);
+    std::string name = field.name;
+    // do not change the case style of the union type field suffix
+    if (is_union_type && name.length() > union_suffix_len)
+      name.erase(name.length() - union_suffix_len, union_suffix_len);
```

Comment:
> it should never happen, this is a safety check; all BASE_TYPE_UTYPE fields should have this suffix - that's how they are created

Please use FLATBUFFERS_ASSERT for expected contract checks, e.g:
```cpp
if (is_union_type) {
  FLATBUFFERS_ASSERT(name.length() > union_suffix_len);
   name.erase(name.length() - union_suffix_len, union_suffix_len); // lets it thow if the first arg is wrong
}
```

Reply:
resolved in latest commit